pypdf/generic/_base.py

```diff
@@ -282,9 +282,27 @@ def indirect_reference(self) -> "IndirectObject":  # type: ignore[override]
 
     def get_object(self) -> Optional["PdfObject"]:
         obj = self.pdf.get_object(self)
-        if obj is None:
-            return None
-        return obj.get_object()
+        return obj
+
+    def __deepcopy__(self, memo: Any) -> "IndirectObject":
```

Comment:
Where do we use this?

Reply:
this is to cope with test_transformation_equivalence if __deepcopy__ is not defined the test fails